apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,173 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// BackendPolicy defines policies associated with backends. For the purpose of
+// this API, a backend is defined as any resource that a route can forward
+// traffic to. A common example of a backend is a Service. Configuration that is
+// implementation specific may be represented with similar implementation
+// specific custom resources.
+type BackendPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BackendPolicySpec   `json:"spec,omitempty"`
+	Status BackendPolicyStatus `json:"status,omitempty"`
```

Comment:
xref'ing kubebuilder tag for status defaulting: https://github.com/kubernetes-sigs/service-apis/pull/324/files?file-filters%5B%5D=.go&file-filters%5B%5D=.html&file-filters%5B%5D=.md&file-filters%5B%5D=.yaml#diff-2927d10c91558c2273bc3a8ea6464b85R34

Reply:
Daneyon, should we resort to a per backend status instead?
https://github.com/kubernetes-sigs/service-apis/pull/268#discussion_r487252302

Reply:
I don't think we can have meaningful defaults here since we'd have to have a new "unknown" default value for each Backend. Although that might be ideal, I don't think we can accomplish it with kubebuilder tags.